### PR TITLE
Everflow for Chassis/T2 topology

### PR DESCRIPTION
**Pull request by abdosi**
What I did:
Enable Eveflow test case infrastructure to work for T2/Multi-Dut topology 

How I did:

This PR takes some of the changes from the PR: https://github.com/sonic-net/sonic-mgmt/pull/6958 and build on top of it.
Here is the summary of the infrastructure changes:

- Concept of `everflow_dut` and `remote_dut` per everflow monitor port direction. Everflow Dut represents entity where given flow will ingress and get mirrored out. Remote Dut represents entity where the Mirror session IP is directly reachable via static route
- setup_info has concept of `upstream dut` and `downstream dut` which will be same for Single Dut Scenario and 2 different for chassis use case. This two dut will be mapped as `everflow_dut` and `remote_dut` 
- Added `ingress_router_mac` and `egress_router_mac`  per everflow monitor port direction. This is needed as each ASIC,LC in case of `chassis-packet` can have differeent router-mac
- TTL check is enhanced to match exact value in mirror packet. For T2 exp TTL is Configured TTL -1 , For Multi-asic T1 it is Configured TTL - 2 and for rest it will be Configured TTL
- All the 3 test cases enhanced to use `setup_info` fixture.
- Made sure topology has min of 2 Peer Neighbors in given namespace for test case execution
- Removed code for parsing of command `show mirror_session` for getting monitor port. Monitor port will always from come from `setup_info`
- Source Port is randomize in given namespace to get different entropy across multiple run of the test case.

How I verify:
Manually Verified on both T0,  T1 topology and T2 (`chassis-packet`) topology

### CODE REVIEW

**Comment by abdosi** [comment]
cc @vperumal

**Comment by judyjoseph** [comment]
@sanmalho-git, everflow test update for optimizations and to support chassis-packet

**Comment by abdosi** [comment]
@sanmalho-git please take this PR and have another PR on top of it for VOQ Chassis.
cc @judyjoseph

**Comment by abdosi** [comment]
@vperumal Please start using this PR.

File: tests/everflow/everflow_test_utilities.py [hunk 1]
```diff
@@ -47,111 +48,148 @@
...
+    upstream_acl_capabilities = upstream_acl_capability_facts["acl_capabilities"]
+
+    downstream_switch_capabilities = downstream_switch_capability_facts["switch_capabilities"]["switch"]
+    downstream_acl_capabilities = downstream_acl_capability_facts["acl_capabilities"]
+
 
-    test_mirror_v4 = switch_capabilities["MIRROR"] == "true"
-    test_mirror_v6 = switch_capabilities["MIRRORV6"] == "true"
+    test_mirror_v4 = upstream_switch_capabilities["MIRROR"] == "true" and downstream_switch_capabilities["MIRROR"] == "true"
+    test_mirror_v6 = upstream_switch_capabilities["MIRRORV6"] == "true" and  downstream_switch_capabilities["MIRRORV6"] == "true"
```
Comment (bingwang-ms) [inline tests/everflow/everflow_test_utilities.py:150]: For T0 device, what's the `downstream_switch_capabilities["MIRROR"]` ?

File: tests/everflow/everflow_test_utilities.py [hunk 2]
```diff
@@ -247,18 +291,62 @@ def get_port_info(in_port_list, out_port_list, out_port_ptf_id_list, out_port_la
...
+    setup_information = gen_setup_information(downstream_duthost, upstream_duthost, tbinfo)
+
     # Disable BGP so that we don't keep on bouncing back mirror packets
     # If we send TTL=1 packet we don't need this but in multi-asic TTL > 1
-    duthost.command("sudo config bgp shutdown all")
+
+    if 't2' in topo:
+        for duthost in duthosts.frontend_nodes:
+            duthost.command("sudo config bgp shutdown all")
+            duthost.command("mkdir -p {}".format(DUT_RUN_DIR))
```
Comment (bingwang-ms) [inline tests/everflow/everflow_test_utilities.py:332]: Is it necessary to put `duthost.command("mkdir -p {}".format(DUT_RUN_DIR))` in the for loop?

File: tests/everflow/everflow_test_utilities.py [hunk 3]
```diff
@@ -247,18 +291,62 @@ def get_port_info(in_port_list, out_port_list, out_port_ptf_id_list, out_port_la
...
-    duthost.command("mkdir -p {}".format(DUT_RUN_DIR))
 
     yield setup_information
 
     # Enable BGP again
-    duthost.command("sudo config bgp startup all")
+    if 't2' in topo:
+        for duthost in duthosts.frontend_nodes:
+            duthost.command("sudo config bgp startup all")
+            duthost.command("rm -rf {}".format(DUT_RUN_DIR))
```
Comment (bingwang-ms) [inline tests/everflow/everflow_test_utilities.py:345]: Same as above, should we put `duthost.command("rm -rf {}".format(DUT_RUN_DIR))` outside the for loop?

File: tests/everflow/everflow_test_utilities.py [hunk 4]
```diff
@@ -406,17 +501,19 @@ def setup_mirror_session(self, duthosts, rand_one_dut_hostname, config_method):
         Yields:
             dict: Information about the mirror session configuration.
         """
-        duthost = duthosts[rand_one_dut_hostname]
-        session_info = BaseEverflowTest.mirror_session_info("test_session_1", duthost.facts["asic_type"])
+        duthost_set = BaseEverflowTest.get_duthost_set(setup_info)
 
-        BaseEverflowTest.apply_mirror_config(duthost, session_info, config_method)
+        for duthost in duthost_set:
+            session_info = BaseEverflowTest.mirror_session_info("test_session_1", duthost.facts["asic_type"])
```
Comment (bingwang-ms) [inline tests/everflow/everflow_test_utilities.py:507]: Please put `session_info = BaseEverflowTest.mirror_session_info("test_session_1", duthost.facts["asic_type"])` outside the for loop.

File: tests/everflow/everflow_test_utilities.py [hunk 5]
```diff
@@ -273,7 +361,7 @@ def add_route(duthost, prefix, nexthop, namespace):
         namespace: namsespace/asic to add the route
 
     """
-    duthost.shell(duthost.get_vtysh_cmd_for_namespace("vtysh -c \"configure terminal\" -c \"ip route {} {}\"".format(prefix, nexthop), namespace))
+    duthost.shell(duthost.get_vtysh_cmd_for_namespace("vtysh -c \"configure terminal\" -c \"ip route {} {} tag 1\"".format(prefix, nexthop), namespace))
```
Comment (bingwang-ms) [inline tests/everflow/everflow_test_utilities.py:364]: Why do we need `tag 1` here?

**Comment by bingwang-ms** [comment]
Please also verify this change in T0 and T1 testbed. Thanks

**Comment by bingwang-ms** [comment]
azp run

File: tests/everflow/everflow_test_utilities.py (hunk 5 above)
Comment (abdosi) [reply tests/everflow/everflow_test_utilities.py:364]: @bingwang-ms  : Check this: https://github.com/sonic-net/sonic-mgmt/pull/4721

File: tests/everflow/everflow_test_utilities.py (hunk 1 above)
Comment (abdosi) [reply tests/everflow/everflow_test_utilities.py:150]: for T0 upStream and downStreamDut are same so both point to same capabilities. This is just to keep code geneic and not have `if..else`

File: tests/everflow/everflow_test_utilities.py (hunk 2 above)
Comment (abdosi) [reply tests/everflow/everflow_test_utilities.py:332]: yes, we need to create DUT_RUN_DIR on both everflow dut and remote dut. We send traffic on both dut to test mirroring happening on same dut and also across dut.

File: tests/everflow/everflow_test_utilities.py (hunk 3 above)
Comment (abdosi) [reply tests/everflow/everflow_test_utilities.py:345]: same as above

File: tests/everflow/everflow_test_utilities.py (hunk 4 above)
Comment (abdosi) [reply tests/everflow/everflow_test_utilities.py]: added check to invoke only one time

**Comment by abdosi** [comment]
@bingwang-ms Have address all of your comments. please check.

**Comment by abdosi** [comment]
@bingwang-ms can you please check and approve

**Pull request merged**